File: spring-cloud-load-balancing-feign/user/src/main/java/com/github/jccode/springcloud/loadbalancingfeign/client/SayHelloClient.java
package com.github.jccode.springcloud.loadbalancingfeign.client;

import org.springframework.cloud.netflix.feign.FeignClient;
import org.springframework.web.bind.annotation.RequestMapping;

@FeignClient(name = "say-hello", fallback = SayHelloClientFallback.class)
public interface SayHelloClient {

    @RequestMapping("/greeting")
    String greet();


}

File: spring-cloud-load-balancing-feign/user/src/main/scala/com/github/jccode/springcloud/loadbalancingfeign/client/SayHelloClient.scala
package com.github.jccode.springcloud.loadbalancingfeign.client

import org.springframework.cloud.netflix.feign.FeignClient
import org.springframework.web.bind.annotation.RequestMapping

@FeignClient(name = "say-hello", fallback = classOf[SayHelloClientFallback])
trait SayHelloClient {

  @RequestMapping(Array("/greeting"))
  def greet: String
  
}
